### DISCUSSION TITLE
Error undefined: No tag found in ref or input!

### DISCUSSION THREAD

**Discussion by TomCockram** [community/community#26381] [Actions]
<p>Hi,</p>
<p>I am trying to create an action that will detect when a change has happened on a certain branch and then build a release. The problem I’m getting is that in my file I use this line:</p>
<p><code>ncipollo/release-action@v1</code> This requires a tag to be available otherwise it will fail. I have created and pushed a tag but when I run the action I get this error</p>
<p><code>Error undefined: No tag found in ref or input!</code></p>
<p>This is my yml file:</p>
<pre><code class="lang-auto">name: Manual workflow
on:  
  push:
    branches:
      - dev
jobs:
  build:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        node-version: [12.x]

    steps:
      - uses: actions/checkout@v2.3.4

      - name: Cache node modules
        uses: actions/cache@v2.1.3
        with:
          path: ~/.npm
          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
          restore-keys: |
            ${{ runner.os }}-node-
      - name: Node ${{ matrix.node-version }}
        uses: actions/setup-node@v1.4.4
        with:
          node-version: ${{ matrix.node-version }}
      - name: npm install and     
        run: |
          npm i
          npm run-script build --prod  
      - name: Push Build to Releases
        uses: ncipollo/release-action@v1
        with:
          artifacts: "dist/angular-githubaction/*"
          token: ${{ secrets.TOKEN }}
</code></pre>

**Comment by airtower-luna** [comment]
<p>Looks like the problem is that your workflow runs on branch push, not tag push, so <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#github-context"><code>github.ref</code></a> doesn’t contain your tag.</p>

**Comment by TomCockram** [comment]
<p>Hmm, how would I get it to include my tags? As far as I’m aware you can set GitHub actions to perform when something is pushed to a certain branch.</p>
<p>I have added<br>
<code> - uses: actions/checkout@v2         with:           fetch-depth: 0</code><br>
this so that it will fetch everything but that hasn’t fixed my problem.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>It’s not about the job, it’s about the workflow trigger. You currently have this:</p>
<aside class="quote no-group" data-username="TomCockram" data-post="1" data-topic="143883">
<div class="title">
<div class="quote-controls"></div>
 TomCockram:</div>
<blockquote>
<pre><code class="lang-auto">on:  
  push:
    branches:
      - dev
</code></pre>
</blockquote>
</aside>
<p>That means the workflow will run if you push the <code>dev</code> branch, and <code>github.ref</code> will always be <code>refs/heads/dev</code>. The workflow will <strong>not</strong> run when you push a tag, so that action cannot automatically determine the tag you want to use.</p>
<p>According to the <a href="https://github.com/ncipollo/release-action/blob/299f7ef0883a9248608af73f077481c47401fa08/README.md"><code>ncipollo/release-action</code> README</a> you have two options:</p>
<ul>
<li>Run the workflow on tag pushes. This is the obvious one: If you push a tag, the release will be made using that tag. For example, to make a release on any tag:</li>
</ul>
<pre><code class="lang-yaml">on:  
  push:
    tags:
      - '*'
</code></pre>
<ul>
<li>Specify the tag name when calling the action, by adding a <code>tag</code> keyword in the <code>with</code> section.</li>
</ul>
<p>If you want to run on both tag and branch pushes and skip the release on non-tag pushes, you could use an <code>if</code> condition like</p>
<pre><code class="lang-yaml">if: ${{ startsWith(github.ref, 'refs/tags/') }}
</code></pre>
<p>on the release step to run it only for tags.</p>

**Reply by SatheeshVxceed** [reply]
Its works for me

**Comment by TomCockram** [comment]
<p>Thanks, I ended up just changing it to a trigger on tag</p>
